packages/cli/src/handlers/download.ts

```diff
@@ -17,79 +21,34 @@ const DOWNLOAD_FOLDER = 'lightdash';
 export type DownloadHandlerOptions = {
     verbose: boolean;
 };
-export const downloadHandler = async (
-    options: DownloadHandlerOptions,
-): Promise<void> => {
-    GlobalState.setVerbose(options.verbose);
-    await checkLightdashVersion();
 
-    const config = await getConfig();
-    if (!config.context?.apiKey || !config.context.serverUrl) {
-        throw new AuthorizationError(
-            `Not logged in. Run 'lightdash login --help'`,
-        );
-    }
+const dumpIntoFiles = async (
+    folder: 'charts' | 'dashboards',
+    items: (ChartAsCode | DashboardAsCode)[],
+) => {
+    const outputDir = path.join(process.cwd(), DOWNLOAD_FOLDER, folder);
 
-    const projectId = config.context.project;
-    if (!projectId) {
-        throw new Error(
-            'No project selected. Run lightdash config set-project',
-        );
-    }
-    const chartsAsCode = await lightdashApi<
-        ApiChartAsCodeListResponse['results']
-    >({
-        method: 'GET',
-        url: `/api/v1/projects/${projectId}/charts/code`,
-        body: undefined,
-    });
-    console.info(`Downloading ${chartsAsCode.length} charts`);
+    console.info(`Writting ${items.length} ${folder} into ${outputDir}`);
+    // Make directory
+    const created = await fs.mkdir(outputDir, { recursive: true });
+    if (created) console.info(`Created new folder: ${outputDir} `);
 
-    const outputDir = path.join(process.cwd(), DOWNLOAD_FOLDER);
-    console.info(`Creating new path for files on ${outputDir} `);
-
-    try {
-        await fs.mkdir(outputDir, { recursive: true });
-    } catch (error) {
-        // Directory already exists
-    }
-
-    for (const chart of chartsAsCode) {
-        const chartPath = path.join(outputDir, `${chart.slug}.yml`);
-        GlobalState.debug(`> Writing chart to ${chartPath}`);
-        const chartYml = yaml.dump(chart, {
+    for (const item of items) {
+        const itemPath = path.join(outputDir, `${item.slug}.yml`);
+        const chartYml = yaml.dump(item, {
             quotingType: '"',
         });
-        await fs.writeFile(chartPath, chartYml);
+        await fs.writeFile(itemPath, chartYml);
     }
-
-    // TODO delete files if chart don't exist ?*/
 };
 
-export const uploadHandler = async (
-    options: DownloadHandlerOptions,
-): Promise<void> => {
-    GlobalState.setVerbose(options.verbose);
-    await checkLightdashVersion();
-
-    const config = await getConfig();
-    if (!config.context?.apiKey || !config.context.serverUrl) {
-        throw new AuthorizationError(
-            `Not logged in. Run 'lightdash login --help'`,
-        );
-    }
+const downloadFiles = async <T extends ChartAsCode | DashboardAsCode>(
```

Comment:
Should this perhaps be called `readCodeFiles`? just so it is not confused with actually downloading the files?